hedera-node/test-clients/SpecCookbook.md

```diff
@@ -28,6 +29,44 @@ The cookbook contains mostly concrete examples; but it is good to keep the princ
 
 ## Patterns
 
+### DO minimize the impact of feature flag rollouts
+
+When adding a feature controlled by a `feature.isEnabled` config, consider three categories of tests:
+1. New tests that require `feature.isEnabled=true` to pass.
+2. New tests that require `feature.isEnabled=false` to pass.
+3. Existing tests that require `feature.isEnabled=false` to pass.
+
+We must ensure that all three categories of tests can run and pass **no matter if the feature flag's default
+value is `true` or `false`**. Tests that assume a default value can create complications if the flag's default value
+needs to be toggled due to tradeoffs in the release schedule, further complicating scheduling challenges.
+
+Let's address each category individually.
+
+First, place every new test that requires `feature.isEnabled=true` in a `@HapiTestLifecycle` test class with a
+`@BeforeAll` method that enables the feature and then restores the default after the test class finishes:
+```java
+@BeforeAll
+static void beforeAll(@NonNull final TestLifecycle testLifecycle) {
+  testLifecycle.overrideInClass(Map.of("feature.isEnabled", "true"));
+}
+```
+(For a large feature there will be multiple such test classes, of course, each with its own `@BeforeAll` method.)
+
+Second, if there are any new tests that require `feature.isEnabled=false`, place them in a test class with a
+`@BeforeAll` method that disables the feature and then restores the default after the test class finishes:
+```java
+@BeforeAll
+static void beforeAll(@NonNull final TestLifecycle testLifecycle) {
+  testLifecycle.overrideInClass(Map.of("feature.isEnabled", "false"));
+}
+```
+
+Third, consider existing tests that require `feature.isEnabled=false` to pass. By default, move all such tests into an
+appropriate `@HapiTestLifecycle` test class as above. The only exception is if there is just one such test; in that
+case, use `@LeakyHapiTest(overrides = {"feature.isEnabled"})` and start the test by overriding the flag to `false`. No 
```

Comment:
Is the reason for this exception simply that it is lighter weight in terms of required syntax?  If so, consider whether we should have this exception _or not_?  Having only _one_ pattern might be easier overall, even if the occasional unitary test has to have a few extra lines of code.

Reply:
Agreed, removed this exception.